src/Collector/RegexCollector.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Collector;
+
+abstract class RegexCollector
+{
+    /**
+     * @param array<string, string|array> $configuration
+     */
+    abstract protected function getPattern(array $configuration): string;
+
+    /**
+     * @param array<string, string|array> $configuration
+     */
+    protected function getValidatedPattern(array $configuration): string
+    {
+        $pattern = $this->getPattern($configuration);
+        set_error_handler(static function () {}, E_WARNING);
```

Comment:
I don't recommend adding new ignore to exist baseline files.

If you want suppress static analysis errors intentionally, you can add inline annotation.

```php
/**
 * @psalm-suppress InvalidArgument 
 * @phpstan-ignore-next-line
 */
set_error_handler(static function () {}, E_WARNING);
```

Reply:
This is currently not feasible because of php-cs-fixer deleting the docbloc

Reply:
>  because of php-cs-fixer deleting the docbloc

oh My bad...